### PR TITLE
feat(new-application): add scaffolding feature for navigation skeleton

### PR DESCRIPTION
**Pull request by jwx**
Add feature scaffolding principles and the navigation skeleton as an option when creating a new custom project.

### CODE REVIEW

File: lib/resources/features/jquery.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+const ProjectItem = require('../../project-item').ProjectItem;
+
+module.exports = function (project, model, options) {
+  project.addToDependencies(
+    'jquery@^2.2.4',
+  );
+  // if (model.bundler.id !== 'webpack') {
```
Comment (JeroenVinke) [inline lib/resources/features/jquery.js]: do we still need this?

File: lib/resources/features/navigation/users.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import {autoinject} from 'aurelia-framework';
+import {HttpClient} from 'aurelia-fetch-client';
+// import 'fetch';
```
Comment (JeroenVinke) [inline lib/resources/features/navigation/users.ts]: can we remove this comment?

**Comment by JeroenVinke** [comment]
I wonder if the file `lib/resources/features/fontAwesome.js` might give us issues with linux

**Comment by huochunpeng** [comment]
FYI, for the eslint rejection, you can use eslint command itself to auto fix many of them.

```
Fixing problems:
  --fix                          Automatically fix problems
  --fix-dry-run                  Automatically fix problems without saving the changes to the file system
```

`npx eslint --fix lib/ spec/` should do the trick.

**Comment by huochunpeng** [comment]
@JeroenVinke is this ready to be merged?

**Comment by JeroenVinke** [comment]
Yes

**Comment by huochunpeng** [comment]
Pls do it :-)

**Comment by JeroenVinke** [comment]
@EisenbergEffect is there any particular reason why you waited to merge this pr?

**Comment by EisenbergEffect** [comment]
@JeroenVinke Sorry! It somehow dropped off my radar. It's completely my fault. Reviewing with intent to merge now.

**Comment by EisenbergEffect** [comment]
Woohoo! This is awesome work everyone!

**Comment by EisenbergEffect** [comment]
@JeroenVinke There's a command to generate all too isn't there? I reviewed code pretty quickly so didn't see that. Can you remind me so I can test that and work on updating our skeletons repo?

**Comment by JeroenVinke** [comment]
Sure, if you run `au generate-skeletons` it creates projects in the current directory using the definitions that are in this directory: 
https://github.com/aurelia/cli/tree/master/lib/commands/generate-skeletons/project-definitions

**Comment by EisenbergEffect** [comment]
Thank you @JeroenVinke I know I've asked you many times. This time I've actually written it down and made it part of my release notes so I can publish the skeletons whenever I publish a CLI update. Thank you everyone for your hard work on this. It's a huge deal for the CLI. Great work!

**Pull request merged**